File: Node-js/2024/2024-001/2024-001-part-001.js
const fs = require('fs');

// File path
const filePath = "\\\\vmware-host\\Shared Folders\\C\\advent-of-code-002\\node-js\\2024\\2024_001\\input.txt";

// Read the file
fs.readFile(filePath, 'utf8', (err, data) => {
    if (err) {
        console.error("Error reading file:", err);
        return;
    }

    // Split the data into lines
    const lines = data.split('\n');

    // Initialize left and right lists
    const leftList = [];
    const rightList = [];

    // Process each line
    lines.forEach(line => {
        const parts = line.trim().split(/\s+/); // Split by whitespace
        if (parts.length === 2) {
            leftList.push(parseInt(parts[0], 10));
            rightList.push(parseInt(parts[1], 10));
        }
    });

    // Sort both lists
    leftList.sort((a, b) => a - b);
    rightList.sort((a, b) => a - b);

    // Calculate the total distance
    let totalDistance = 0;
    for (let i = 0; i < leftList.length; i++) {
        totalDistance += Math.abs(leftList[i] - rightList[i]);
    }

    // Output the total distance
    console.log("Total Distance:", totalDistance);
});

File: other-programming-languages/Ruby/2024/2024-001/2024-001-part-001.rb
# Function to calculate the total distance between two sorted lists
def calculate_total_distance(file_path)
  # Read the file and load the location IDs into two arrays
  left_list = []
  right_list = []

  # Open the file and process each line
  File.foreach(file_path) do |line|
    # Split the line by spaces and convert the values to integers
    left, right = line.split.map(&:to_i)
    left_list << left
    right_list << right
  end

  # Sort both lists
  left_list.sort!
  right_list.sort!

  # Calculate the total distance
  total_distance = 0
  left_list.each_with_index do |left, index|
    right = right_list[index]
    total_distance += (left - right).abs
  end

  # Output the total distance
  puts "Total Distance: #{total_distance}"
end

# File path for input data
file_path = '\\\\vmware-host\\Shared Folders\\C\\advent-of-code-002\\Ruby\\2024\\2024-001\\input.txt'

# Call the function to calculate the total distance
calculate_total_distance(file_path)
